syntax = "proto3";

package yandex.cloud.loadtesting.agent.v1;

option go_package = "agent";


service MonitoringService {
  rpc AddMetric (AddMetricRequest) returns (AddMetricResponse);
}

message AddMetricRequest {
  reserved 2;
  string compute_instance_id = 1;
  string job_id = 3;
  repeated MetricChunk chunks = 4;
}

message MetricChunk {
  repeated Metric data = 1;
  int64 timestamp = 2;
  string comment = 3;
  string instance_host = 4;
}

message Metric {
  string metric_type = 1;
  string metric_name = 2;
  double metric_value = 3;
}

message AddMetricResponse {
  string metric_trail_id = 1;
  int64 code = 2;
}
